Fix colors for tmux 2.9

goronfreeman opened the pull request:
tmux 2.9 removed the `foo-bg`, `foo-fg`, and `foo-attr` options. Here is the relevant line from the changelog:

>The individual -fg, -bg and -attr options have been removed; they were superseded by -style options in tmux 1.9.

This results in broken styles in many places in nord-tmux. I updated the relevant lines to use the new `foo-style` syntax, and now the theme appears how it did in tmux 2.8.

goronfreeman committed:
Fix colors for tmux 2.9

jasondibenedetto commented:
Any update on getting this merged in?

arcticicestudio commented:
Hi @goronfreeman :wave:, thanks for your contribution :+1:
Sorry for the late reply, but unfortunately my free time to spend working on OSS has been very limited in the past few months.

Since this is a breaking change I'll need to evaluate if it is possible to support both version ranges `<=2.8` and `>=2.9`. I know they introduced a "soft deprecation" in tmux v1.9, but there were no warnings when the deprecated attributes were found in a configuration file. Introducing such a breaking change in a minor version (2.9) instead of major (3.0+) is also not the way a large project like tmux shouldn't have gone.
Anyway, I'll try to test if it is possible to still provide support for both version ranges, otherwise Nord must adapt to the 2.9+ properties, raise the minimum required tmux version to 2.9 and document the breaking change in order for users to migrate to the next release version.

goronfreeman commented:
@arcticicestudio Let me know if I can help out with that. Thanks for the details!

arcticicestudio commented:
@goronfreeman Thanks 😄 
Due to the current health situation this world has to fight with I also needed some time “off-screen“ (job, life etc.) and currently processing all the new issues & PRs in other projects, but I've moved this topic higher on my list to ensure other users are not to negatively affected by this. Maybe I'll need help when it comes to testing on other OS so I can save time setting up virtual machines again ^-^

boydkelly commented:
> Hi @goronfreeman , thanks for your contribution 
> Sorry for the late reply, but unfortunately my free time to spend working on OSS has been very limited in the past few months.
> 
> Since this is a breaking change I'll need to evaluate if it is possible to support both version ranges `<=2.8` and `>=2.9`. I know they introduced a "soft deprecation" in tmux v1.9, but there were no warnings when the deprecated attributes were found in a configuration file. Introducing such a breaking change in a minor version (2.9) instead of major (3.0+) is also not the way a large project like tmux shouldn't have gone.
> Anyway, I'll try to test if it is possible to still provide support for both version ranges, otherwise Nord must adapt to the 2.9+ properties, raise the minimum required tmux version to 2.9 and document the breaking change in order for users to migrate to the next release version.

The new syntax is valid since 1.9 as you mention above.   So if the nord syntax is updated the required minimum tmux version is 1.9 and not 2.9.   Cheers!

cubercsl commented:
tmux is 3.2 now, any update for this?

arcticicestudio commented on src/nord.conf:
```diff
@@ -25,17 +25,13 @@ set -g status on
-set -g pane-border-fg black
-set -g pane-active-border-bg black
-set -g pane-active-border-fg brightblack
+set -g pane-border-style bg=black,fg=black
+set -g pane-active-border-style bg=black,fg=brightblack
```
```suggestion
set -g pane-border-style bg=default,fg=brightblack
set -g pane-active-border-style bg=default,fg=blue
```

The style of the borders were not optimal before tmux 2.9 so I've played around with the styles and like to use a thinner line and ensure only the active pane is highlighted visually.

<p align="center"><strong>Before</strong></p>
<p align="center">
  <img src="https://user-images.githubusercontent.com/7836623/122653554-9a809a00-d145-11eb-8552-604f698cd656.png" width="80%">
</p>
<p align="center"><strong>Improved</strong></p>
<p align="center">
  <img src="https://user-images.githubusercontent.com/7836623/122653552-994f6d00-d145-11eb-89c6-06dd27deb9a2.png" width="80%">
</p>

arcticicestudio reviewed (changes requested):
I've added an improvement suggestion for a style that was used even before tmux `v2.9`, otherwise all changes working fine.
The adjustment of the documentations can be resolved in another PR, including in the [Nord docs repository][1].

[1]: https://github.com/arcticicestudio/nord-docs

arcticicestudio committed:
Thinner lines for active pane border

The style of the borders were not optimal before tmux 2.9 so they have
been changed to appear as a thinner line to ensure only the active pane
is highlighted visually.

Co-authored-by: Sven Greb <development@svengreb.de>

GH-34

arcticicestudio reviewed (approved):
Thanks again @goronfreeman and everyone else who contributed and your patience to get this done. 

--- 

Resolves #40

ioogithub commented:
This "breaking change" has existed for almost 2 years now. Any change that this change will be published to the main version so we can fix our systems?
